python/lsst/ts/genericcamera/genericcameracsc.py

```diff
@@ -455,23 +570,162 @@ async def liveView_loop(self, exposure_time):
         self.isLive = False
         self.log.info("liveView_loop - End")
 
+    async def run_auto_exposure_loop(self, min_exp_time, max_exp_time, configuration):
+        """Run the auto exposure capture loop.
+
+        The cadence of the images is determined by the value of
+        AUTO_EXP_IMAGE_INTERVAL.
+
+        Parameters
+        ----------
+        min_exp_time: `float`
+            The minimum exposure time to use.
+        max_exp_time: `float`
+            The maximum exposure time to use.
+        configuration: `str`
+            A Yaml string containing additional configuration
+            parameters.
+        """
+        self.log.info("autoExposure_loop - Start")
+        self.isAutoExposure = True
+
+        shutter = False
+        sensors = ""
+        keyValueMap = ""
+        obsNote = ""
+        if configuration != "":
+            config = yaml.safe_load(configuration)
+            if "shutter" in config:
+                shutter = config["shutter"]
+            if "sensors" in config:
+                sensors = config["sensors"]
+            if "keyValueMap" in config:
+                keyValueMap = config["keyValueMap"]
+            if "obsNote" in config:
+                obsNote = config["obsNote"]
+
+        images_in_sequence = 0
+        image_index = 0
+        exposure_time = min_exp_time
+
+        try:
+            while self.runAutoExposureTask:
```

Comment:
Consider moving this `while` loop to a dedicated method. In fact, consider breaking this into smaller methods.. For instance, this could be:

```
try:
    await self.run_auto_exposure()
except Exception as e:
   ...
```

Reply:
Done.